### PR TITLE
Add GKE, Amazon ECS, and Azure App Service deployment guides

### PR DESCRIPTION
**Pull request by imjohnbo**
### Why:

Google Kubernetes Engine deployment guide as part of a group of select partner deployment guides. 

cc/ @runleonarun, @swinton, @jpadams 👋 

### What's being changed:

* Added a new section to [this page](https://docs.github.com/en/free-pro-team@latest/actions/guides) to include continuous deployment (CD):

<p align="center">

</p>

* Added a new "deployment guide" for Google Kubernetes Engine based on the existing [starter workflow](https://github.com/actions/starter-workflows/blob/master/ci/google.yml).

### Check off the following:
- [ ] All of the tests are passing.
- [ ] I have reviewed my changes in staging. (look for the **deploy-to-heroku** link in your pull request, then click **View deployment**)
- [ ] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [ ] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

**Comment by jpadams** [comment]
@imjohnbo LGTM in the preview environment!
image

**Comment by martin389** [comment]
Hi @imjohnbo 👋 - Thanks for this great guide! I've added some small edits and structural changes to the draft. Could you check that you're happy with my changes?

The rendered preview is here: https://docs-2085--deployment-guidesad.herokuapp.com/en/free-pro-team@latest/actions/guides/deploying-to-google-kubernetes-engine

File: content/actions/guides/deploying-to-google-kubernetes-engine.md [hunk 1]
```diff
@@ -11,68 +11,89 @@ versions:
 {% data reusables.actions.enterprise-github-hosted-runners %}
 
 ### Introduction
-[Google Kubernetes Engine](https://cloud.google.com/kubernetes-engine) (GKE) is a managed Kubernetes cluster service from Google Cloud and is a great option for hosting your containerized workloads in the cloud or on premise.
 
-This guide will show you how to use GitHub Actions to build and deploy a containerized application from Google Container Registry (GCR) to GKE. 
+This guide explains how to use {% data variables.product.prodname_actions %} to build a containerized application in Google Container Registry (GCR) and deploy it to Google Kubernetes Engine (GKE).
```
Comment (imjohnbo) [inline content/actions/guides/deploying-to-google-kubernetes-engine.md:15]: @martin389, what would you think about something like this instead 💭...

> ...{% data variables.product.prodname_actions %} to build a containerized application, push it to Google Container Registry (GCR), and deploy it to Google Kubernetes Engine (GKE)

The build is taken care of by GitHub Actions, and GCR is used an intermediary registry for GKE to pull from.

**Comment by imjohnbo** [comment]
> Hi @imjohnbo 👋 - Thanks for this great guide! I've added some small edits and structural changes to the draft. Could you check that you're happy with my changes?
> 
> The rendered preview is here: https://docs-2085--deployment-guidesad.herokuapp.com/en/free-pro-team@latest/actions/guides/deploying-to-google-kubernetes-engine

This looks great to me, @martin389. Thanks for the quick review. 🙏

**Review by dstiers11 — APPROVED** [review]

**Comment by jpadams** [comment]
Thanks Docs team for the assistance and improvements! 🙌 We'd love to get these published as soon as possible to support the current DevOps campaign and our goal to deliver these three initial deploy guides this quarter to support an ask from Jeremy Epling in https://github.com/github/business-development/issues/980.

cc @kathrynhsu @crichID

File: content/actions/guides/deploying-to-azure-app-service.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
+---
+title: Deploying to Azure App Service Service
```
Comment (runleonarun) [inline content/actions/guides/deploying-to-azure-app-service.md:2]:
```suggestion
title: Deploying to Azure App Service
```

**Comment by runleonarun** [comment]
@jpadams It looks like we need to edit the two newly added guides. Martin is OOO but @lucascosti will so kindly edit them so we can ship ASAP. 💐

File: content/actions/guides/deploying-to-google-kubernetes-engine.md (hunk 1 above)
Comment (lucascosti) [reply content/actions/guides/deploying-to-google-kubernetes-engine.md:15]: Sounds good to me, I'll make the change 👍

**Comment by lucascosti** [comment]
👋 @imjohnbo, @jpadams, @swinton! 

Thanks again for this great content. I've done edits of the Azure and Amazon ECS guide drafts to fit in with our docs style and structure.

Could you please have a look and make sure I haven't changed the meaning of anything?

In particular, for some example code blocks, I changed the user-replaceable values that looked like environment variables (e.g. `$ECR_REPOSITORY`), to be a more obvious a user-replaceable value (e.g. `MY_ECR_REPOSITORY`). If I'm mistaken and these do actually need to be environment variables, please let me know 😅.

Rendered previews can be seen on the stage:

* https://docs-2085--deployment-guidesad.herokuapp.com/en/free-pro-team@latest/actions/guides/deploying-to-amazon-elastic-container-service
* https://docs-2085--deployment-guidesad.herokuapp.com/en/free-pro-team@latest/actions/guides/deploying-to-azure-app-service

@martin389 / @runleonarun I'd love to get a proofread on my changes as well before these are published 🙂 Also note that I haven't personally tested these procedures.

File: content/actions/guides/deploying-to-amazon-elastic-container-service.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+env:
+  AWS_REGION: MY_AWS_REGION                   # set this to your preferred AWS region, e.g. us-west-1
+  ECR_REPOSITORY: MY_ECR_REPOSITORY           # set this to your Amazon ECR repository name
+  ECS_SERVICE: MY-ECS-SERVICE                 # set this to your Amazon ECS service name
+  ECS_CLUSTER: MY-ECS-CLUSTER                 # set this to your Amazon ECS cluster name
+  ECS_TASK_DEFINITION: MY-ECS-TASK-DEFINITION # set this to the path to your Amazon ECS task definition
+                                               # file, e.g. .aws/task-definition.json
+  CONTAINER_NAME: MY-CONTAINER-NAME           # set this to the name of the container in the
+                                               # containerDefinitions section of your task definition
```
Comment (swinton) [inline content/actions/guides/deploying-to-amazon-elastic-container-service.md:85]: Suggestion: standardize on `_`s:

```suggestion
  AWS_REGION: MY_AWS_REGION                   # set this to your preferred AWS region, e.g. us-west-1
  ECR_REPOSITORY: MY_ECR_REPOSITORY           # set this to your Amazon ECR repository name
  ECS_SERVICE: MY_ECS_SERVICE                 # set this to your Amazon ECS service name
  ECS_CLUSTER: MY_ECS_CLUSTER                 # set this to your Amazon ECS cluster name
  ECS_TASK_DEFINITION: MY_ECS_TASK_DEFINITION # set this to the path to your Amazon ECS task definition
                                               # file, e.g. .aws/task-definition.json
  CONTAINER_NAME: MY_CONTAINER_NAME           # set this to the name of the container in the
                                               # containerDefinitions section of your task definition
```

File: content/actions/guides/deploying-to-amazon-elastic-container-service.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+   {% raw %}```bash{:copy}
+   aws ecr create-repository \
+       --repository-name MY_ECR_REPOSITORY \
+       --region MY_AWS_REGION
+   ```{% endraw %}
+
+   Ensure that you use the same Amazon ECR repository name for the `ECR_REPOSITORY` variable in the workflow below.
+
+   Ensure that you use the same AWS region value for the `AWS_REGION` variable in the workflow below.
```
Comment (swinton) [inline content/actions/guides/deploying-to-amazon-elastic-container-service.md:35]: Suggestion: be more explicit about the relationship between variables.

```suggestion
   {% raw %}```bash{:copy}
   aws ecr create-repository \
       --repository-name MY_ECR_REPOSITORY \
       --region MY_AWS_REGION
   ```{% endraw %}

   Ensure that you use the same Amazon ECR repository name (represented here by `MY_ECR_REPOSITORY`) for the `ECR_REPOSITORY` variable in the workflow below.

   Ensure that you use the same AWS region value for the `AWS_REGION` (represented here by `MY_AWS_REGION`) variable in the workflow below.
```

**Review by swinton — APPROVED** [review]
Thanks @lucascosti. A couple of suggestions from me :bow:

File: content/actions/guides/deploying-to-amazon-elastic-container-service.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+   See the documentation for each action used below for the recommended IAM policies for the IAM user, and methods for handling the access key credentials.
+
+### Creating the workflow
+
+Once you've completed the prerequisites, you can proceed with creating the workflow.
+
+The following example workflow demonstrates how to build a container image and push it to Amazon ECR. It then updates the task definition with the new image ID, and deploys the task definition to Amazon ECS.
+
+Ensure that you provide your own values for all the variables in the `env:` key of the workflow.
```
Comment (runleonarun) [inline content/actions/guides/deploying-to-amazon-elastic-container-service.md:67]:
```suggestion
Ensure that you provide your own values for all the variables in the `env` key of the workflow.
```

**Review by runleonarun — APPROVED** [review]
Found a tiny typo otherwise this is ready to ship!

**Comment by runleonarun** [comment]
@jpadams @swinton and @imjohnbo We can merge this today. It looks good! Thanks for the team work! 👥

**Review by iCG-SE — COMMENTED** [review]
# this config file contains all config fields with comments
# NOTE: this is not a particularly useful config file
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
# patch the generated kubeadm config with some extra settings
kubeadmConfigPatches:
- |
  apiVersion: kubelet.config.k8s.io/v1beta1
  kind: KubeletConfiguration
  evictionHard:
    nodefs.available: "0%"
# patch it further using a JSON 6902 patch
kubeadmConfigPatchesJSON6902:
- group: kubeadm.k8s.io
  version: v1beta2
  kind: ClusterConfiguration
  patch: |
    - op: add
      path: /apiServer/certSANs/-
      value: my-hostname
# 1 control plane node and 3 workers
nodes:
# the control plane node config
- role: control-plane
# the three workers
- role: worker
- role: worker
- role: worker

**Pull request merged**